plugins/discovery-ec2/src/main/java/org/elasticsearch/discovery/ec2/AwsEc2ServiceImpl.java

```diff
@@ -49,20 +51,50 @@
     private AmazonEC2 buildClient(Ec2ClientSettings clientSettings) {
         final AWSCredentialsProvider credentials = buildCredentials(logger, clientSettings);
         final ClientConfiguration configuration = buildConfiguration(logger, clientSettings);
-        final AmazonEC2 client = buildClient(credentials, configuration);
+
+        final AwsClientBuilder.EndpointConfiguration endpointConfiguration;
         if (Strings.hasText(clientSettings.endpoint)) {
             logger.debug("using explicit ec2 endpoint [{}]", clientSettings.endpoint);
-            client.setEndpoint(clientSettings.endpoint);
+            endpointConfiguration = new AwsClientBuilder.EndpointConfiguration(clientSettings.endpoint,
```

Comment:
As I understand it, the region and endpoint has to go together and really specifying the endpoint should not be done. The region is used to determine signing as well, which means that using a wrong region for an endpoint is unlikely to work (AFAIK). Given that we calculate the region from the endpoint anyway, we may as well just call `AwsClientBuilder.withRegion` with the region, this will ensure both the endpoint and signing region are set.